sentry_sdk/client.py

```diff
@@ -456,10 +456,11 @@ def _should_sample_error(
         event,  # type: Event
     ):
         # type: (...) -> bool
-        not_in_sample_rate = (
-            self.options["sample_rate"] < 1.0
-            and random.random() >= self.options["sample_rate"]
-        )
+        try:
+            sample_rate = self.options["issues_sampler"](event)
```

Comment:
Would be good to wrap the actual call in `capture_internal_exceptions`, see how we do it for `before_send` [here](https://github.com/getsentry/sentry-python/blob/4d10edfe7233d5adc2ceeeb984d8f93dfa3a29eb/sentry_sdk/client.py#L381-L382), which is also a user-provided callable.

You'll probably need to split this into two parts: checking if `self.options["issues_sampler"]` is set, and actually calling it, and only wrap the latter in `capture_internal_exceptions`.